labscript_devices/DummyPseudoclock/labscript_devices.py

```diff
@@ -15,24 +15,90 @@
 # and labscript. The device is a PseudoclockDevice, and can be the sole device
 # in a connection table or experiment.
 
-import labscript_utils.h5_lock
-import h5py
-from labscript import PseudoclockDevice, Pseudoclock, ClockLine
+from labscript import PseudoclockDevice, Pseudoclock, ClockLine, config, LabscriptError
+import numpy as np
 
 
 class DummyPseudoclock(PseudoclockDevice):
 
     description = 'Dummy pseudoclock'
-    clock_limit = 1e6
-    clock_resolution = 1e-6
+    clock_limit = 10e6
+    clock_resolution = 25e-9
+    clock_type = 'fast clock'
+    trigger_delay = 350e-9
+    wait_delay = 2.5e-6
+    allowed_children = None
```

Comment:
This should be `allowed_children = [PseudoClock]`. Setting to `None` means any subclass of `Device` can be a direct child.

Reply:
You may also wish to also copy across the `add_device` method from the PineBlaster to stop more than one `Pseudoclock` from being added as a child to the `PsedoclockDevice`.

Reply:
Protecting against more than one `ClockLine` being attached requires duplicating the `PineBlasterPseudoclock` subclass and using that instead of `Psuedoclock` when instantiating the child pseudoclock.